java/src/jmri/jmrit/jython/JynstrumentFactory.java

```diff
@@ -66,21 +67,25 @@ public static Jynstrument createInstrument(String path, Object context) {
     // validate Jynstrument path, return className
     private static String validate(String path) {
         if (path == null) {
+            log.error("Path is null");
             return null;
         }
         if (path.length() - 4 < 0) {
-            log.error("File name too short");
+            log.error("File name too short (should at least end with .jyn) (got {})", path);
             return null;
         }
+        if (path.endsWith(File.separator)) {
+            path = path.substring(0, path.length()-File.separator.length());
+        }
         File f = new File(path);
 
         // Path must be a folder named xyz.jin
         if (!f.isDirectory()) {
             log.debug("Not a directory, trying parent");
             return null;
         }
-        if (path.substring(path.length() - 4).compareToIgnoreCase(".jyn") != 0) {
-            log.debug("Not an instrument");
+        if (! path.toLowerCase().startsWith(".jyn", path.length() - 4)) {
```

Comment:
Shouln't this be:
`if (! path.toLowerCase().endsWith(".jyn")) {`
instead?

Reply:
yes, definitively, pushing a correction.